extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -1016,22 +1091,66 @@ static class StringTransfer implements TransferObject<Binary[]> {
 
         private final ChunkSource.GetContext context;
         private ObjectChunk<String, Values> chunk;
-        private final Binary[] buffer;
+        private Binary[] buffer;
         private final ColumnSource<?> columnSource;
 
+        /**
+         * Stores the maximum number of values in a single page
+         */
+        private final int maxValuesPerPage;
+
+        /**
+         * Stores the maximum size of data in a single page. For strings, we don't know in advance what the length of
+         * each entry would be, therefore we need to keep track of both maximum size and number of values.
+         */
+        private final int targetPageSize;
+
+        /**
+         * Total number of chunks buffered so far
+         */
+        private int numChunksBuffered;
```

Comment:
This name doesn't seem correct.  Isn't this a chunk index?  maybe `currentChunkIdx` is more clear?